lib/sisimai/bite/email/x3.rb

```diff
@@ -0,0 +1,141 @@
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
+        blanklines = 0      # (Integer) The number of blank lines
+        readcursor = 0      # (Integer) Points the current cursor position
+        recipients = 0      # (Integer) The number of 'Final-Recipient' header
+        v = nil
+
+        hasdivided.each do |e|
+          if readcursor.zero?
+            # Beginning of the bounce message or delivery status part
+            if e =~ Re1[:begin]
+              readcursor |= Indicators[:deliverystatus]
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] == 0
+            # Beginning of the original message part
+            if e =~ Re1[:rfc822]
+              readcursor |= Indicators[:'message-rfc822']
+              next
+            end
+          end
+
+          if readcursor & Indicators[:'message-rfc822'] > 0
+            # After "message/rfc822"
+            if e.empty?
+              blanklines += 1
+              break if blanklines > 1
+              next
+            end
+            rfc822list << e
+
+          else
+            # Before "message/rfc822"
+            next if readcursor & Indicators[:deliverystatus] == 0
+            next if e.empty?
+
+            # ============================================================================
+            #      This is an automatically generated Delivery Status Notification.
+            #
+            # Delivery to the following recipients failed permanently:
+            #
+            #   * kijitora@example.com
+            #
+            #
+            # ============================================================================
+            #                             Technical details:
+            #
+            # SMTP:RCPT host 192.0.2.8: 553 5.3.0 <kijitora@example.com>... No such user here
+            #
+            #
+            # ============================================================================
+            v = dscontents[-1]
+
+            if cv = e.match(/\A[ \t]+[*][ ]([^ ]+[@][^ ]+)\z/)
+              #   * kijitora@example.com
+              if v['recipient']
+                # There are multiple recipient addresses in the message body.
+                dscontents << Sisimai::Bite.DELIVERYSTATUS
+                v = dscontents[-1]
+              end
+              v['recipient'] = cv[1]
+              recipients += 1
+
+            else
+              # Detect error message
+              if cv = e.match(/\ASMTP:([^ ]+)[ ](.+)\z/)
+                # SMTP:RCPT host 192.0.2.8: 553 5.3.0 <kijitora@example.com>... No such user here
+                v['command'] = cv[1].upcase
+                v['diagnosis'] = cv[2]
+
+              elsif cv = e.match(/\ARouting: (.+)/)
+                # Routing: Could not find a gateway for kijitora@example.co.jp
+                v['diagnosis'] = cv[1]
+              end
+            end
+          end
+        end
```

Comment:
__[rubocop]__
* Style/IfInsideElse: Convert `if` nested inside `else` to `elsif`.

<!-- Comment by [SideCI](https://sideci.com) -->